qiskit/circuit/quantumcircuit.py

```diff
@@ -1436,6 +1405,8 @@ def __deepcopy__(self, memo=None):
         result._data.replace_bits(
             qubits=_copy.deepcopy(self._data.qubits, memo),
             clbits=_copy.deepcopy(self._data.clbits, memo),
+            qregs=_copy.deepcopy(self._data.qregs, memo),
+            cregs=_copy.deepcopy(self._data.cregs, memo),
```

Comment:
Not for this PR necessarily, but I feel like this should all be handled by `_data.copy(deepcopy=True)` now. With everything living in rust now, it should be easier to just bake this into the `CircuitData.copy()` implementation (the whole `replace_bits()` call not just the register piece added here).